client/src/pages/UserAppUpload/UserAppUpload.js

```diff
@@ -0,0 +1,424 @@
+                                        }
+                                        onClose={createOrganisationModal.hide}
+                                    />
+                                )}
+                                <ReactFinalForm.Field
+                                    required
+                                    name="developerOrganisation"
+                                    label="Organisation"
+                                    placeholder="Select an organisation"
+                                    component={SingleSelectFieldFF}
+                                    className={styles.organisationField}
+                                    validate={hasValue}
+                                    options={organisationOptions}
+                                />
+                                <Button
+                                    secondary
+                                    onClick={createOrganisationModal.show}
+                                >
+                                    Create organisation
+                                </Button>
+                            </div>
+                        </section>
+
+                        <section className={styles.formSection}>
+                            <h3 className={styles.subheader}>
+                                Upload app file (Required)
+                            </h3>
+                            <p className={styles.description}>
+                                App files should be submitted as zip.
+                            </p>
+                            <ReactFinalForm.Field
+                                required
+                                name="file"
+                                buttonLabel="Upload a zip file"
+                                accept=".zip"
+                                component={FileInputFieldFF}
+                                className={styles.field}
+                                validate={hasValue}
+                            />
+                        </section>
+
+                        <section className={styles.formSection}>
+                            <h3 className={styles.subheader}>
+                                Upload logo (Required)
+                            </h3>
+                            <p className={styles.description}>
+                                A logo will be displayed in the App Hub
+                                listings. A good-looking logo will help users
+                                find your app.
+                            </p>
+                            <ReactFinalForm.Field
+                                required
+                                name="logo"
+                                buttonLabel="Upload a logo"
+                                accept="image/*"
+                                component={FileInputFieldFF}
+                                className={styles.field}
+                                validate={hasValue}
```

Comment:
The backend checks that the mime type starts with `image/` so pretty much any image file format is supported. Upload size is a bit trickier since we set a maximum multipart form size of 20MB, which for this page would include the bundle size. When uploading images separately the maximum size is 20MB (which I think we should reduce to ~5MB) so it's not exactly intuitive for users. I guess we could put something like "2MB maximum file size" in the help text as a kind of soft limit?

Reply:
Yeah, I think that would work. I think a help text with the common image formats would help avoid the most common mistake (trying to upload a PDF). So in that case I'd go for `Only .jpg, .png and .gif files, 2MB max size`.